ssg/oval.py

```diff
@@ -106,24 +106,27 @@ def replace_external_vars(tree):
     return tree
 
 
+def find_testfile_or_exit(testfile):
+    try:
+        return find_testfile(testfile)
+    except RuntimeError as exc:
+        print(
+            "ERROR: %s Please specify a valid OVAL file."
+            % str(exc), file=sys.stderr)
+        sys.exit(1)
+
+
 def find_testfile(testfile):
     """Find OVAL files in CWD or shared/oval"""
     for path in ['.', SHARED_OVAL]:
         for root, _, _ in os.walk(path):
-            searchfile = root + '/' + testfile
-            if not os.path.isfile(searchfile):
-                searchfile = ""
-            else:
-                testfile = searchfile.strip()
-                # Most likely found file, exit this loop
-                break
-
-    if not os.path.isfile(testfile):
-        print("ERROR: %s does not exist! Please specify a valid OVAL file."
-              % testfile, file=sys.stderr)
-        sys.exit(1)
+            searchfile = os.path.join(root, testfile).strip()
+            if os.path.isfile(searchfile):
+                return searchfile
 
-    return testfile
+    # If we have not returned, we haven't found anything
+    msg = "%s does not exist!" % testfile
+    raise RuntimeError(msg)
```

Comment:
This probably should be an `IOError`.

Reply:
I took a look at [Python3 docs](https://docs.python.org/3/library/exceptions.html#IOError), and `IOError` seems to be appropriate when the OS can't read from a file, as it is an alias to `OSError`. However, I gave it another thought and written the code in a way that it doesn't use exceptions.